### PR TITLE
Original Generate API supports role

### PR DESCRIPTION
**Pull request by FrostML**
### PR types

New features

### PR changes

APIs

### Description

Original Generate API supports role ids. 
Fix FasterGeneration generate without role ids.

### CODE REVIEW

File: paddlenlp/transformers/generation_utils.py [hunk 1]
```diff
@@ -390,6 +390,10 @@ def expand_inputs_for_generation(input_ids,
             model_kwargs["encoder_output"] = paddle.gather(encoder_output,
                                                            index)
 
+        if "role_ids" in model_kwargs and model_kwargs["role_ids"] is not None:
+            role_ids = model_kwargs["role_ids"]
+            model_kwargs["role_ids"] = paddle.gather(role_ids, index)
+
```
Comment (guoshengCS) [inline paddlenlp/transformers/generation_utils.py:396]: 现在是否和`FasterUnifiedTransformer.expand_inputs_for_generation`一致了， 那直接把那个去掉吧

File: paddlenlp/transformers/generation_utils.py [hunk 2]
```diff
@@ -448,6 +453,12 @@ def update_model_kwargs_for_generation(outputs,
                     axis=-1)
             model_kwargs["attention_mask"] = attention_mask
 
+        # update role_ids
+        if "role_ids" in model_kwargs and model_kwargs["role_ids"] is not None:
+            role_ids = model_kwargs["role_ids"]
+            model_kwargs["role_ids"] = paddle.concat(
+                [role_ids, role_ids[:, -1:]], axis=-1)
+
```
Comment (guoshengCS) [inline paddlenlp/transformers/generation_utils.py:461]: @smallv0221  也再check下吧

File: paddlenlp/transformers/generation_utils.py (hunk 1 above)
Comment (FrostML) [reply paddlenlp/transformers/generation_utils.py:396]: Done. Thanks.

**Review by guoshengCS — APPROVED** [review]

**Review by smallv0221 — APPROVED** [review]
LGTM

**Pull request merged**